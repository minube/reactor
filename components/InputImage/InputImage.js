import { string } from 'prop-types';
import React, { PureComponent } from 'react';
import { View } from 'react-native';

import Image from '../Image';
import Input from '../Input';
import styles from './InputImage.style';

class InputImage extends PureComponent {
  static propTypes = {
    value: string,
  };

  static defaultProps = {
    value: '',
  };

  render() {
    const { props: { value, ...inherit } } = this;
    const isValidUrl = value.length > 0;

    return (
      <View style={[styles.container, inherit.style]}>
        { isValidUrl && <Image source={{ uri: value }} style={styles.image} /> }
        <Input
          {...inherit}
          style={[styles.input, isValidUrl && styles.inputMargin]}
          value={value}
        />
      </View>
    );
  }
}

export default InputImage;
