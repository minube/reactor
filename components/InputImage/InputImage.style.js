import { StyleSheet } from 'react-native';

import { LAYOUT, THEME } from '../../common';

const {
  BORDER_RADIUS, COLOR, SPACE, UNIT,
} = THEME;

export default StyleSheet.create({
  container: {
    ...LAYOUT.STYLE.ROW,
    alignItems: 'flex-end',
  },

  input: {
    flex: 1,
  },

  inputMargin: {
    marginLeft: SPACE.XS,
  },

  image: {
    width: UNIT * 4.4,
    height: UNIT * 4.4,
    borderRadius: BORDER_RADIUS,
    backgroundColor: COLOR.BASE,
  },

  imageContainer: {
    marginBottom: SPACE.REGULAR,
  },
});
