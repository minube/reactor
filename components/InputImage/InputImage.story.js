import React from 'react';
import { action } from '@storybook/addon-actions';
import { storiesOf } from '@storybook/react';
import { array, text, boolean, number, object, select } from '@storybook/addon-knobs/react';

import InputImage from './InputImage';

const VALUE = 'http://soyjavi.com/assets/images/soyjavi.png';
const STYLE = { backgroundColor: 'rgba(0,255,0,0.25)', padding: 10, width: 512 };

storiesOf('🛠 InputImage', module)
  .addWithJSX('default', () => (
    <InputImage />
  ))
  .addWithJSX('disabled', () => (
    <InputImage disabled />
  ))
  .addWithJSX('error', () => (
    <InputImage error="Required field" />
  ))
  .addWithJSX('hint', () => (
    <InputImage hint="Optional field" />
  ))
  .addWithJSX('label', () => (
    <InputImage label="Username" />
  ))
  .addWithJSX('value', () => (
    <InputImage value="three" value={VALUE} />
  ))
  .addWithJSX('⚡ onChange', () => (
    <InputImage onChange={action('InputImage.onChange()')} title="Press me" value={VALUE} />
  ))
  .addWithJSX('style', () => (
    <InputImage style={STYLE} value={VALUE} />
  ))
  .addWithJSX('🏀 Playground', () => (
    <InputImage
      disabled={boolean('disabled', false)}
      error={text('error', null)}
      hint={text('hint', null)}
      label={text('label', null)}
      value={text('value', VALUE)}
      style={object('style', STYLE)}
    />
  ))

