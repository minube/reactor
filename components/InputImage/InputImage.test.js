import React from 'react';
import renderer from 'react-test-renderer';

import InputImage from './InputImage';

const VALUE = 'http://soyjavi.com/assets/images/soyjavi.jpg';
const STYLE = { backgroundColor: 'rgba(0,255,0,0.25)', padding: 10 };

describe('<InputImage>', () => {
  it('renders', () => {
    const tree = renderer.create(<InputImage />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {disabled}', () => {
    const tree = renderer.create(<InputImage disabled />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {error}', () => {
    const tree = renderer.create(<InputImage error="Hello World" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {hint}', () => {
    const tree = renderer.create(<InputImage hint="Hello World" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {label}', () => {
    const tree = renderer.create(<InputImage label="Hello World" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {required}', () => {
    const tree = renderer.create(<InputImage required />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {value}', () => {
    const tree = renderer.create(<InputImage value={VALUE} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {style}', () => {
    const tree = renderer.create(<InputImage style={STYLE} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
