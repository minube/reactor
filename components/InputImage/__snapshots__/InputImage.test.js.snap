// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputImage> renders 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "marginBottom": 20,
        },
        Array [
          Object {
            "flex": 1,
          },
          false,
        ],
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderColor": "#cdd2d4",
            "borderRadius": 2.5,
            "borderWidth": 1,
            "display": "flex",
            "flexDirection": "row",
            "paddingHorizontal": 5,
          },
          false,
          false,
          undefined,
        ]
      }
    >
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        autoCorrect={false}
        blurOnSubmit={true}
        editable={true}
        keyboardType="default"
        multiline={false}
        onBlur={[Function]}
        onFocus={[Function]}
        placeholderTextColor="#707070"
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
              "color": "#4a4a4a",
              "flex": 1,
              "fontFamily": undefined,
              "fontSize": 16,
              "fontWeight": "400",
              "minHeight": 42,
              "paddingHorizontal": 5,
              "paddingVertical": 7.5,
              "width": "100%",
            },
            false,
          ]
        }
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
  </View>
</View>
`;

exports[`<InputImage> when {disabled} 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "marginBottom": 20,
        },
        Array [
          Object {
            "flex": 1,
          },
          false,
        ],
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderColor": "#cdd2d4",
            "borderRadius": 2.5,
            "borderWidth": 1,
            "display": "flex",
            "flexDirection": "row",
            "paddingHorizontal": 5,
          },
          Object {
            "backgroundColor": "#f4f4f4",
          },
          false,
          false,
        ]
      }
    >
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        autoCorrect={false}
        blurOnSubmit={true}
        editable={false}
        keyboardType="default"
        multiline={false}
        onBlur={[Function]}
        onFocus={[Function]}
        placeholderTextColor="#707070"
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
              "color": "#4a4a4a",
              "flex": 1,
              "fontFamily": undefined,
              "fontSize": 16,
              "fontWeight": "400",
              "minHeight": 42,
              "paddingHorizontal": 5,
              "paddingVertical": 7.5,
              "width": "100%",
            },
            Object {
              "color": "#707070",
            },
          ]
        }
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
  </View>
</View>
`;

exports[`<InputImage> when {error} 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "marginBottom": 20,
        },
        Array [
          Object {
            "flex": 1,
          },
          false,
        ],
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderColor": "#cdd2d4",
            "borderRadius": 2.5,
            "borderWidth": 1,
            "display": "flex",
            "flexDirection": "row",
            "paddingHorizontal": 5,
          },
          false,
          false,
          Object {
            "borderColor": "#d33c44",
          },
        ]
      }
    >
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        autoCorrect={false}
        blurOnSubmit={true}
        editable={true}
        keyboardType="default"
        multiline={false}
        onBlur={[Function]}
        onFocus={[Function]}
        placeholderTextColor="#707070"
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
              "color": "#4a4a4a",
              "flex": 1,
              "fontFamily": undefined,
              "fontSize": 16,
              "fontWeight": "400",
              "minHeight": 42,
              "paddingHorizontal": 5,
              "paddingVertical": 7.5,
              "width": "100%",
            },
            false,
          ]
        }
        underlineColorAndroid="transparent"
        value=""
      />
      <Image
        resizeMode="contain"
        source={
          Object {
            "testUri": "../../../components/Icon/assets/error.png",
          }
        }
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "height": 20,
              "width": 20,
            },
            Object {
              "height": 20,
              "marginHorizontal": 5,
              "width": 20,
            },
            undefined,
          ]
        }
      />
    </View>
  </View>
</View>
`;

exports[`<InputImage> when {hint} 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "marginBottom": 20,
        },
        Array [
          Object {
            "flex": 1,
          },
          false,
        ],
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderColor": "#cdd2d4",
            "borderRadius": 2.5,
            "borderWidth": 1,
            "display": "flex",
            "flexDirection": "row",
            "paddingHorizontal": 5,
          },
          false,
          false,
          undefined,
        ]
      }
    >
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        autoCorrect={false}
        blurOnSubmit={true}
        editable={true}
        keyboardType="default"
        multiline={false}
        onBlur={[Function]}
        onFocus={[Function]}
        placeholderTextColor="#707070"
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
              "color": "#4a4a4a",
              "flex": 1,
              "fontFamily": undefined,
              "fontSize": 16,
              "fontWeight": "400",
              "minHeight": 42,
              "paddingHorizontal": 5,
              "paddingVertical": 7.5,
              "width": "100%",
            },
            false,
          ]
        }
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      numberOfLines={2}
      style={
        Array [
          Object {
            "color": "#4a4a4a",
            "fontFamily": undefined,
            "fontWeight": "400",
            "textAlignVertical": "center",
          },
          Object {
            "fontSize": 12,
            "lineHeight": 16,
          },
          Object {
            "color": "#707070",
          },
          Object {
            "marginTop": 4,
            "maxWidth": "100%",
            "paddingLeft": 10,
          },
        ]
      }
    >
      Hello World
    </Text>
  </View>
</View>
`;

exports[`<InputImage> when {label} 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "marginBottom": 20,
        },
        Array [
          Object {
            "flex": 1,
          },
          false,
        ],
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      numberOfLines={1}
      style={
        Array [
          Object {
            "color": "#4a4a4a",
            "fontFamily": undefined,
            "fontWeight": "400",
            "textAlignVertical": "center",
          },
          Object {
            "fontFamily": undefined,
            "fontSize": 14,
            "fontWeight": "600",
            "lineHeight": 16,
          },
          Object {
            "color": "#707070",
          },
          Object {
            "marginBottom": 6,
            "marginLeft": 10,
          },
        ]
      }
    >
      Hello World
    </Text>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderColor": "#cdd2d4",
            "borderRadius": 2.5,
            "borderWidth": 1,
            "display": "flex",
            "flexDirection": "row",
            "paddingHorizontal": 5,
          },
          false,
          false,
          undefined,
        ]
      }
    >
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        autoCorrect={false}
        blurOnSubmit={true}
        editable={true}
        keyboardType="default"
        multiline={false}
        onBlur={[Function]}
        onFocus={[Function]}
        placeholderTextColor="#707070"
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
              "color": "#4a4a4a",
              "flex": 1,
              "fontFamily": undefined,
              "fontSize": 16,
              "fontWeight": "400",
              "minHeight": 42,
              "paddingHorizontal": 5,
              "paddingVertical": 7.5,
              "width": "100%",
            },
            false,
          ]
        }
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
  </View>
</View>
`;

exports[`<InputImage> when {required} 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "marginBottom": 20,
        },
        Array [
          Object {
            "flex": 1,
          },
          false,
        ],
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderColor": "#cdd2d4",
            "borderRadius": 2.5,
            "borderWidth": 1,
            "display": "flex",
            "flexDirection": "row",
            "paddingHorizontal": 5,
          },
          false,
          false,
          undefined,
        ]
      }
    >
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        autoCorrect={false}
        blurOnSubmit={true}
        editable={true}
        keyboardType="default"
        multiline={false}
        onBlur={[Function]}
        onFocus={[Function]}
        placeholderTextColor="#707070"
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
              "color": "#4a4a4a",
              "flex": 1,
              "fontFamily": undefined,
              "fontSize": 16,
              "fontWeight": "400",
              "minHeight": 42,
              "paddingHorizontal": 5,
              "paddingVertical": 7.5,
              "width": "100%",
            },
            false,
          ]
        }
        underlineColorAndroid="transparent"
        value=""
      />
      <Image
        resizeMode="contain"
        source={
          Object {
            "testUri": "../../../components/Icon/assets/error_outline.png",
          }
        }
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "height": 20,
              "width": 20,
            },
            Object {
              "height": 20,
              "marginHorizontal": 5,
              "width": 20,
            },
            undefined,
          ]
        }
      />
    </View>
  </View>
</View>
`;

exports[`<InputImage> when {style} 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
      },
      Object {
        "backgroundColor": "rgba(0,255,0,0.25)",
        "padding": 10,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "marginBottom": 20,
        },
        Array [
          Object {
            "flex": 1,
          },
          false,
        ],
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderColor": "#cdd2d4",
            "borderRadius": 2.5,
            "borderWidth": 1,
            "display": "flex",
            "flexDirection": "row",
            "paddingHorizontal": 5,
          },
          false,
          false,
          undefined,
        ]
      }
    >
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        autoCorrect={false}
        blurOnSubmit={true}
        editable={true}
        keyboardType="default"
        multiline={false}
        onBlur={[Function]}
        onFocus={[Function]}
        placeholderTextColor="#707070"
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
              "color": "#4a4a4a",
              "flex": 1,
              "fontFamily": undefined,
              "fontSize": 16,
              "fontWeight": "400",
              "minHeight": 42,
              "paddingHorizontal": 5,
              "paddingVertical": 7.5,
              "width": "100%",
            },
            false,
          ]
        }
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
  </View>
</View>
`;

exports[`<InputImage> when {value} 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignContent": "center",
          "alignItems": "center",
          "backgroundColor": "#F5F5F6",
          "justifyContent": "center",
        },
        Object {
          "backgroundColor": "#cdd2d4",
          "borderRadius": 3,
          "height": 44,
          "marginBottom": 1,
          "width": 44,
        },
      ]
    }
  >
    <Image
      onLoad={[Function]}
      source={
        Object {
          "uri": "http://soyjavi.com/assets/images/soyjavi.jpg",
        }
      }
      style={
        Object {
          "backgroundColor": "#cdd2d4",
          "borderRadius": 3,
          "height": 44,
          "marginBottom": 1,
          "width": 44,
        }
      }
    />
    <ActivityIndicator
      animating={true}
      color="white"
      hidesWhenStopped={true}
      size="large"
      style={
        Array [
          Object {
            "position": "absolute",
          },
          false,
        ]
      }
    />
  </View>
  <View
    style={
      Array [
        Object {
          "marginBottom": 20,
        },
        Array [
          Object {
            "flex": 1,
          },
          Object {
            "marginLeft": 10,
          },
        ],
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderColor": "#cdd2d4",
            "borderRadius": 2.5,
            "borderWidth": 1,
            "display": "flex",
            "flexDirection": "row",
            "paddingHorizontal": 5,
          },
          false,
          false,
          undefined,
        ]
      }
    >
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        autoCorrect={false}
        blurOnSubmit={true}
        editable={true}
        keyboardType="default"
        multiline={false}
        onBlur={[Function]}
        onFocus={[Function]}
        placeholderTextColor="#707070"
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
              "color": "#4a4a4a",
              "flex": 1,
              "fontFamily": undefined,
              "fontSize": 16,
              "fontWeight": "400",
              "minHeight": 42,
              "paddingHorizontal": 5,
              "paddingVertical": 7.5,
              "width": "100%",
            },
            false,
          ]
        }
        underlineColorAndroid="transparent"
        value="http://soyjavi.com/assets/images/soyjavi.jpg"
      />
    </View>
  </View>
</View>
`;
